types/passport/lib/errors/authenticationerror.d.ts

```diff
@@ -0,0 +1,5 @@
+export default class AuthenticationError extends Error {
```

Comment:
Why does this need its own file? Is it intended for users to import from their code? Otherwise it can just go in index.d.ts

Reply:
The linked file says
```
 * @api private
```
which really makes me believe this is not intended to be exposed

Reply:
This is a type that can be thrown by passport if you set the option `failWithError: true`.

If you try to do `err instanceof AuthenticationError` in an error handler you will need to do the next in vanilla node js:
```
const AuthenticationError = require('passport/lib/errors/authenticationerror');

if (err instanceof AuthenticationError) {
  // Handle passport errors
} else {
  // Handle other errors
}
```

With these changes, you'll be able to do that same comparison in TS without compiler errors. If there is a cleaner way to do it please let me know.
```
import AuthenticationError from 'passport/lib/errors/authenticationerror';

if (err instanceof AuthenticationError) {
  // Handle passport errors
} else {
  // Handle other errors
}
```